File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int count = 1;
        int currM = nums[0];
        for(int i = 1; i<nums.length;i++){
           if(currM == nums[i]){
            count++;
           }else{
            count--;
           }

           if(count == 0){
            currM = nums[i];
            count++;
           }
        }
        return currM;
    }
}

File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        curr_maj = nums[0]
        count = 0
        for i in nums:
            if i == curr_maj:
                count+=1
            else:
                count-=1
            
            if count == 0:
                curr_maj = i
                count = 1
        return curr_maj